store/postgres/src/relational.rs

```diff
@@ -572,6 +573,49 @@ impl Layout {
         Ok(entities_for_type)
     }
 
+    pub fn find_changes(
+        &self,
+        conn: &PgConnection,
+        block: BlockNumber,
+    ) -> Result<Vec<EntityModification<EntityType>>, StoreError> {
+        let mut tables = Vec::new();
+        for table in self.tables.values() {
+            tables.push(&**table);
+        }
+
+        let inserts_or_updates =
+            FindChangesQuery::new(&self.catalog.site.namespace, &tables[..], block)
+                .load::<EntityData>(conn)?;
+        let deletes = FindDeletesQuery::new(&self.catalog.site.namespace, &tables[..], block)
+            .load::<EntityData>(conn)?;
+
+        let mut changes = Vec::new();
+
+        for data in deletes.into_iter() {
+            // The complexity is bad here (we're iterating over the set of
+            // updated entities for every deletion). This query, however, is
+            // not particularly performance-sensitive
+            if !inserts_or_updates
+                .iter()
+                .any(|update| update.entity_type() == data.entity_type())
```

Comment:
I don't understand the logic here - `entity_type()` is something like `Account` or `Token`, but the comparison tries to find which entities were truly deleted rather than updated (delete + insert) The comparison therefore would also have to take the entity's id into account.

Reply:
I was under the impression that, for a given block number and entity type, you'd have at most **one** row that is either updated/inserted/deleted. I.e. you can't have two entities with the same entity type deleted at the same block. That would mean that we don't need entity ID comparisons.

Reply:
Indeed, block ranges for any given entity don't overlap, so we'd never have more than a single entity per entity type here. Am I missing something?

Reply:
> I was under the impression that, for a given block number and entity type, you'd have at most one row that is either updated/inserted/deleted.

Any number of entities can be deleted / updated / inserted in a block.

Reply:
Indeed, but you can't have two entities that are deleted / updated / inserted in a block with the same entity type, or can you?

Reply:
Ah, no. Entities are either deleted or updated/inserted but not both in the same block. (At least not when looking at the database. They may be deleted and recreated when processing the block but that would not show here because these operations are rolled into one before transacting to the database.)

Reply:
But if it's either or, then you can just assume that the two sets (updates, deletes) are disjoint and you don't need the inner loop. Instead you should be able to take all deletions as they are, no?

Reply:
Addressed with proper documentation, comments, and performing lookups of duplicates with a `HashSet`.